src/Orleans/Streams/QueueAdapters/IBatchContainer.cs

```diff
@@ -54,9 +54,11 @@ public interface IBatchContainer
         StreamSequenceToken SequenceToken { get; }
 
         /// <summary>
-        /// The RequestContext that was set at the time event was generated and enqueued into the persistent provider.
+        /// Gives an opportunity to IBatchContainer to set any data in the RequestContext before this IBatchContainer is sent to consumers.
+        /// It can be the data that was set at the time event was generated and enqueued into the persistent provider or any other data.
         /// </summary>
-        Dictionary<string, object> RequestContext { get; }
+        /// <returns>True if the RequestContext was indeed modified, false otherwise.</returns>
+        bool SetRequestContext();
```

Comment:
`SetRequestContext` has an unclear meaning to me: it could indicate that we are modifying the `IBatchContainer`, setting its request context. Is there a more suitable name which we could use?

How about `ImportRequestContext`? It fits with the `RequesContext.Import()` call in `AzureQueueBatchContainer`.